helix-core/src/test/java/org/apache/helix/integration/rebalancer/CrushRebalancers/TestCrushAutoRebalance.java

```diff
@@ -298,7 +298,7 @@ private void validateZoneAndTagIsolation(IdealState is, ExternalView ev, int exp
       Assert.assertEquals(assignedZones.size(), expectedReplica);
     }
   }
-
+/* These blank test would cause @AfterClass not get invoked.
```

Comment:
I don't think so. It is delayed but not ignored. Is it a concern if the after class is invoked with some delay?

If so, then all the tests that are developed with a parent test class have this potential issue. Then we need to fix it with a different strategy.

Reply:
You are right. It is delayed after some other test class run. Still, this means hard to track leaking threads, as we want to have all test method in one test class to run at one batch.

Reply:
confirm parent testZone not going to run this way.

Reply:
>Start zookeeper at localhost:2183 in thread main
START TestCrushAutoRebalance at Tue Oct 20 20:43:11 PDT 2020
START TestCrushAutoRebalance testZoneIsolationWithInstanceTag at Tue Oct 20 20:43:13 PDT 2020
END TestCrushAutoRebalance testZoneIsolationWithInstanceTag at Tue Oct 20 20:43:15 PDT 2020, took: 1811ms.
START TestCrushAutoRebalance testZoneIsolationWithInstanceTag at Tue Oct 20 20:43:15 PDT 2020
END TestCrushAutoRebalance testZoneIsolationWithInstanceTag at Tue Oct 20 20:43:16 PDT 2020, took: 1773ms.
START TestCrushAutoRebalance testZoneIsolationWithInstanceTag at Tue Oct 20 20:43:16 PDT 2020
END TestCrushAutoRebalance testZoneIsolationWithInstanceTag at Tue Oct 20 20:43:18 PDT 2020, took: 1718ms.
START TestCrushAutoRebalance testZoneIsolation at Tue Oct 20 20:43:18 PDT 2020
END TestCrushAutoRebalance testZoneIsolation at Tue Oct 20 20:43:20 PDT 2020, took: 1663ms.
START TestCrushAutoRebalance testZoneIsolation at Tue Oct 20 20:43:20 PDT 2020
END TestCrushAutoRebalance testZoneIsolation at Tue Oct 20 20:43:21 PDT 2020, took: 1731ms.
START TestCrushAutoRebalance testZoneIsolation at Tue Oct 20 20:43:21 PDT 2020
END TestCrushAutoRebalance testZoneIsolation at Tue Oct 20 20:43:23 PDT 2020, took: 1672ms.
START TestCrushAutoRebalance testLackEnoughLiveRacks at Tue Oct 20 20:43:23 PDT 2020
END TestCrushAutoRebalance testLackEnoughLiveRacks at Tue Oct 20 20:43:25 PDT 2020, took: 1817ms.
START TestCrushAutoRebalance testLackEnoughRacks at Tue Oct 20 20:43:25 PDT 2020
END TestCrushAutoRebalance testLackEnoughRacks at Tue Oct 20 20:43:27 PDT 2020, took: 1819ms.
END TestCrushAutoRebalance at Tue Oct 20 20:43:27 PDT 2020

Reply:
This actually works.